VSTS Build #49 succeded but it actually failed to build any code

VSTS build 49 with commit 3b36b54a6ccf81eec434ad24c5efe549945563d7 and issue #65  succeded but in reality it failed to use .net core SDK 1.0, and it failed to recognize this failure because the return code returned by cake's `StartProcess` that calls `dotnet build`  is being ignored.

The following is the build's log:

```
2017-03-12T16:03:52.0345918Z ##[section]Starting: Build
2017-03-12T16:03:52.0502158Z Current agent version: '2.112.0'
2017-03-12T16:03:52.7545649Z Downloading task: PowerShell
2017-03-12T16:03:53.1830701Z Downloading task: Cake
2017-03-12T16:03:53.3493837Z Downloading task: PublishTestResults
2017-03-12T16:03:55.5561363Z ##[section]Starting: Get Sources
2017-03-12T16:03:55.6754051Z Syncing repository: carlosmunozrodriguez/minicloner (GitHub)
2017-03-12T16:03:55.6754051Z Prepending Path environment variable with directory containing 'git.exe'.
2017-03-12T16:03:55.7225363Z ##[command]git version
2017-03-12T16:03:56.0697101Z ##[command]git init "D:\a\1\s"
2017-03-12T16:03:56.1478353Z Initialized empty Git repository in D:/a/1/s/.git/
2017-03-12T16:03:56.1478353Z ##[command]git remote add origin https://github.com/carlosmunozrodriguez/minicloner.git
2017-03-12T16:03:56.1790923Z ##[command]git config gc.auto 0
2017-03-12T16:03:56.2103428Z ##[command]git config --get-all http.https://github.com/carlosmunozrodriguez/minicloner.git.extraheader
2017-03-12T16:03:56.2415873Z ##[command]git config --get-all http.proxy
2017-03-12T16:03:56.2884634Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ********" fetch --tags --prune --progress origin
2017-03-12T16:03:57.1582068Z remote: Counting objects: 588, done.        
2017-03-12T16:03:57.1582068Z remote: Compressing objects:   6% (1/16)           
2017-03-12T16:03:57.1582068Z remote: Compressing objects:  12% (2/16)           
2017-03-12T16:03:57.1582068Z remote: Compressing objects:  18% (3/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  25% (4/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  31% (5/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  37% (6/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  43% (7/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  50% (8/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  56% (9/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  62% (10/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  68% (11/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  75% (12/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  81% (13/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  87% (14/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects:  93% (15/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects: 100% (16/16)           
2017-03-12T16:03:57.1737630Z remote: Compressing objects: 100% (16/16), done.        
2017-03-12T16:03:57.1894058Z Receiving objects:   0% (1/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   1% (6/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   2% (12/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   3% (18/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   4% (24/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   5% (30/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   6% (36/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   7% (42/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   8% (48/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:   9% (53/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  10% (59/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  11% (65/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  12% (71/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  13% (77/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  14% (83/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  15% (89/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  16% (95/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  17% (100/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  18% (106/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  19% (112/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  20% (118/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  21% (124/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  22% (130/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  23% (136/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  24% (142/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  25% (147/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  26% (153/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  27% (159/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  28% (165/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  29% (171/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  30% (177/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  31% (183/588)   
2017-03-12T16:03:57.1894058Z Receiving objects:  32% (189/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  33% (195/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  34% (200/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  35% (206/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  36% (212/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  37% (218/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  38% (224/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  39% (230/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  40% (236/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  41% (242/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  42% (247/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  43% (253/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  44% (259/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  45% (265/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  46% (271/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  47% (277/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  48% (283/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  49% (289/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  50% (294/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  51% (300/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  52% (306/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  53% (312/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  54% (318/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  55% (324/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  56% (330/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  57% (336/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  58% (342/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  59% (347/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  60% (353/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  61% (359/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  62% (365/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  63% (371/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  64% (377/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  65% (383/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  66% (389/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  67% (394/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  68% (400/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  69% (406/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  70% (412/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  71% (418/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  72% (424/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  73% (430/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  74% (436/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  75% (441/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  76% (447/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  77% (453/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  78% (459/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  79% (465/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  80% (471/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  81% (477/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  82% (483/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  83% (489/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  84% (494/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  85% (500/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  86% (506/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  87% (512/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  88% (518/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  89% (524/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  90% (530/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  91% (536/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  92% (541/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  93% (547/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  94% (553/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  95% (559/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  96% (565/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  97% (571/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  98% (577/588)   
2017-03-12T16:03:57.2056060Z Receiving objects:  99% (583/588)   
2017-03-12T16:03:57.2056060Z Receiving objects: 100% (588/588)   
2017-03-12T16:03:57.2056060Z Receiving objects: 100% (588/588), 90.55 KiB | 0 bytes/s, done.
2017-03-12T16:03:57.2056060Z remote: Total 588 (delta 1), reused 0 (delta 0), pack-reused 570        
2017-03-12T16:03:57.2056060Z Resolving deltas:   0% (0/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:   1% (4/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:   2% (7/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:   3% (11/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:   6% (21/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:   7% (25/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:   8% (30/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:   9% (32/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  10% (38/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  11% (40/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  12% (42/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  13% (46/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  14% (52/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  15% (54/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  16% (56/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  17% (61/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  22% (79/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  23% (82/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  28% (99/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  29% (102/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  31% (108/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  36% (127/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  39% (137/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  44% (154/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  47% (165/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  50% (174/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  51% (179/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  52% (184/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  54% (188/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  55% (193/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  56% (195/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  60% (209/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  61% (215/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  62% (218/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  63% (220/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  64% (225/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  65% (228/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  66% (231/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  68% (237/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  69% (242/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  70% (246/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  71% (248/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  76% (266/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  84% (293/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  85% (298/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  87% (305/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  88% (308/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  89% (313/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  90% (315/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  92% (322/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  93% (324/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  96% (335/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  97% (341/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas:  99% (346/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas: 100% (348/348)   
2017-03-12T16:03:57.2056060Z Resolving deltas: 100% (348/348), done.
2017-03-12T16:03:57.2518898Z From https://github.com/carlosmunozrodriguez/minicloner
2017-03-12T16:03:57.2518898Z  * [new branch]      develop                -> origin/develop
2017-03-12T16:03:57.2518898Z  * [new branch]      feature/upgrade-vs2017 -> origin/feature/upgrade-vs2017
2017-03-12T16:03:57.2518898Z  * [new branch]      master                 -> origin/master
2017-03-12T16:03:57.2831402Z ##[command]git checkout --progress --force 86e2b8f2f474becaa48ea31160556623838e788f
2017-03-12T16:03:57.3300153Z Note: checking out '86e2b8f2f474becaa48ea31160556623838e788f'.
2017-03-12T16:03:57.3300153Z 
2017-03-12T16:03:57.3300153Z You are in 'detached HEAD' state. You can look around, make experimental
2017-03-12T16:03:57.3300153Z changes and commit them, and you can discard any commits you make in this
2017-03-12T16:03:57.3300153Z state without impacting any branches by performing another checkout.
2017-03-12T16:03:57.3300153Z 
2017-03-12T16:03:57.3300153Z If you want to create a new branch to retain commits you create, you may
2017-03-12T16:03:57.3300153Z do so (now or later) by using -b with the checkout command again. Example:
2017-03-12T16:03:57.3300153Z 
2017-03-12T16:03:57.3300153Z   git checkout -b <new-branch-name>
2017-03-12T16:03:57.3300153Z 
2017-03-12T16:03:57.3300153Z HEAD is now at 86e2b8f... Merge pull request #65 from carlosmunozrodriguez/feature/upgrade-vs2017
2017-03-12T16:03:57.3466208Z ##[section]Finishing: Get Sources
2017-03-12T16:03:57.3466208Z ##[section]Starting: dotnet --info
2017-03-12T16:03:57.3925169Z ==============================================================================
2017-03-12T16:03:57.3925169Z Task         : PowerShell
2017-03-12T16:03:57.3925169Z Description  : Run a PowerShell script
2017-03-12T16:03:57.3925169Z Version      : 1.2.2
2017-03-12T16:03:57.3925169Z Author       : Microsoft Corporation
2017-03-12T16:03:57.3925169Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613736)
2017-03-12T16:03:57.3925169Z ==============================================================================
2017-03-12T16:03:57.5331437Z ##[command]. 'C:\Users\buildguest\AppData\Local\Temp\9b137484-4a8d-46a7-be4b-afd21dde7b16.ps1' 
2017-03-12T16:03:59.8751787Z .NET Command Line Tools (1.0.0-preview2-1-003177)
2017-03-12T16:03:59.8751787Z 
2017-03-12T16:03:59.8751787Z Product Information:
2017-03-12T16:03:59.8751787Z  Version:            1.0.0-preview2-1-003177
2017-03-12T16:03:59.8751787Z  Commit SHA-1 hash:  a2df9c2576
2017-03-12T16:03:59.8751787Z 
2017-03-12T16:03:59.8751787Z Runtime Environment:
2017-03-12T16:03:59.8751787Z  OS Name:     Windows
2017-03-12T16:03:59.8751787Z  OS Version:  6.3.9600
2017-03-12T16:03:59.8751787Z  OS Platform: Windows
2017-03-12T16:04:00.0855402Z  RID:         win81-x64
2017-03-12T16:04:00.3667946Z ##[section]Finishing: dotnet --info
2017-03-12T16:04:00.3667946Z ##[section]Starting: Build
2017-03-12T16:04:00.3667946Z ==============================================================================
2017-03-12T16:04:00.3667946Z Task         : Cake
2017-03-12T16:04:00.3667946Z Description  : Build with Cake
2017-03-12T16:04:00.3667946Z Version      : 0.2.4
2017-03-12T16:04:00.3667946Z Author       : Patrik Svensson
2017-03-12T16:04:00.3667946Z Help         : [More Information about Cake](http://cakebuild.net)
2017-03-12T16:04:00.3667946Z ==============================================================================
2017-03-12T16:04:00.3824137Z Preparing task execution handler.
2017-03-12T16:04:03.7684153Z Executing the powershell script: D:\a\_tasks\Cake_b88ea9a0-7d6e-11e5-b5de-d57d652482f7\0.2.4\Cake.ps1
2017-03-12T16:04:04.7931591Z Restoring packages...
2017-03-12T16:04:04.7931591Z 
2017-03-12T16:04:04.7931591Z 
2017-03-12T16:04:05.5930210Z Installing 'Cake 0.18.0'.
2017-03-12T16:04:05.9212255Z Successfully installed 'Cake 0.18.0'.
2017-03-12T16:04:05.9367740Z Executing build script...
2017-03-12T16:04:05.9367740Z 
2017-03-12T16:04:05.9367740Z 
2017-03-12T16:04:05.9680265Z D:\a\1\s\tools\Cake\Cake.exe "D:\a\1\s\build.cake" -target="Default" -verbosity="Verbose" --paths_tools="D:\a\1\s\tools" 
2017-03-12T16:04:06.3117817Z Analyzing build script...
2017-03-12T16:04:06.3274108Z Processing build script...
2017-03-12T16:04:06.3430621Z Downloading and installing Roslyn...
2017-03-12T16:04:06.5833461Z Installing packages (using https://packages.nuget.org/api/v2)...
2017-03-12T16:04:08.4530038Z Copying files...
2017-03-12T16:04:08.4530038Z Copying Roslyn.Compilers.CSharp.dll...
2017-03-12T16:04:08.4530038Z Copying Roslyn.Compilers.dll...
2017-03-12T16:04:08.4530038Z Deleting installation directory...
2017-03-12T16:04:08.8662637Z Compiling build script...
2017-03-12T16:04:12.6318061Z 
2017-03-12T16:04:12.6318061Z ========================================
2017-03-12T16:04:12.6318061Z Restore
2017-03-12T16:04:12.6318061Z ========================================
2017-03-12T16:04:12.6318061Z Executing task: Restore
2017-03-12T16:04:13.2067547Z warn : The folder 'D:\a\1\s' does not contain a project to restore.
2017-03-12T16:04:13.2536528Z Finished executing task: Restore
2017-03-12T16:04:13.2536528Z 
2017-03-12T16:04:13.2536528Z ========================================
2017-03-12T16:04:13.2536528Z Build
2017-03-12T16:04:13.2536528Z ========================================
2017-03-12T16:04:13.2536528Z Executing task: Build
2017-03-12T16:04:13.3317505Z Specify --help for a list of available options and commands.
2017-03-12T16:04:13.3630013Z ##[error]Unrecognized option '--verbosity'
2017-03-12T16:04:13.3630013Z Finished executing task: Build
2017-03-12T16:04:13.3630013Z 
2017-03-12T16:04:13.3630013Z ========================================
2017-03-12T16:04:13.3630013Z Test
2017-03-12T16:04:13.3630013Z ========================================
2017-03-12T16:04:13.3630013Z Executing task: Test
2017-03-12T16:04:13.8006244Z ##[error]dotnet-test Error: 0 : System.InvalidOperationException: test/Minicloner.Tests/Minicloner.Tests.csproj\project.json does not exist.
2017-03-12T16:04:13.8006244Z ##[error]   at Microsoft.DotNet.Tools.Test.TestCommand.GetProjectPath(String projectPath)
2017-03-12T16:04:13.8006244Z ##[error]   at Microsoft.DotNet.Tools.Test.TestCommand.DoRun(String[] args)
2017-03-12T16:04:13.8006244Z Finished executing task: Test
2017-03-12T16:04:13.8006244Z 
2017-03-12T16:04:13.8006244Z ========================================
2017-03-12T16:04:13.8006244Z Default
2017-03-12T16:04:13.8006244Z ========================================
2017-03-12T16:04:13.8006244Z Executing task: Default
2017-03-12T16:04:13.8006244Z Finished executing task: Default
2017-03-12T16:04:13.8161323Z 
2017-03-12T16:04:13.8161323Z Task                          Duration            
2017-03-12T16:04:13.8161323Z --------------------------------------------------
2017-03-12T16:04:13.8161323Z Restore                       00:00:00.6283429    
2017-03-12T16:04:13.8161323Z Build                         00:00:00.0900428    
2017-03-12T16:04:13.8161323Z Test                          00:00:00.4641506    
2017-03-12T16:04:13.8161323Z Default                       00:00:00.0002892    
2017-03-12T16:04:13.8161323Z --------------------------------------------------
2017-03-12T16:04:13.8161323Z Total:                        00:00:01.1828255    
2017-03-12T16:04:13.8630069Z ##[section]Finishing: Build
2017-03-12T16:04:13.8630069Z ##[section]Starting: Publish Test Results
2017-03-12T16:04:13.8786336Z ==============================================================================
2017-03-12T16:04:13.8786336Z Task         : Publish Test Results
2017-03-12T16:04:13.8786336Z Description  : Publish Test Results to VSTS/TFS
2017-03-12T16:04:13.8786336Z Version      : 1.0.36
2017-03-12T16:04:13.8786336Z Author       : Microsoft Corporation
2017-03-12T16:04:13.8786336Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613742)
2017-03-12T16:04:13.8786336Z ==============================================================================
2017-03-12T16:04:13.8786336Z Preparing task execution handler.
2017-03-12T16:04:14.3473885Z Executing the powershell script: D:\a\_tasks\PublishTestResults_0b0f01ed-7dde-43ff-9cbb-e48954daf9b1\1.0.36\PublishTestResults.ps1
2017-03-12T16:04:14.9411471Z Publishing Test Results...
2017-03-12T16:04:16.5922341Z ##[warning]Failed to publish test results: Could not find a part of the path 'D:\a\1\s\test\Minicloner.Tests\TestResults\TestResults.trx'.
2017-03-12T16:04:16.6234855Z ##[section]Finishing: Publish Test Results
2017-03-12T16:04:16.6234855Z ##[section]Starting: Post Job Cleanup
2017-03-12T16:04:16.6234855Z Cleaning any cached credential from repository: carlosmunozrodriguez/minicloner (Git)
2017-03-12T16:04:16.6391096Z ##[command]git remote set-url origin https://github.com/carlosmunozrodriguez/minicloner.git
2017-03-12T16:04:16.6703599Z ##[command]git remote set-url --push origin https://github.com/carlosmunozrodriguez/minicloner.git
2017-03-12T16:04:16.7016092Z ##[section]Finishing: Post Job Cleanup
2017-03-12T16:04:16.7172342Z ##[section]Finishing: Build

```
Although it looks like it succeeded

![Build #49](https://i.stack.imgur.com/gLAFr.png)

The following needs to be done:

1. Change `build.cake` to actually fail when StartProcess returns an return code different than 0.
    After this change the build should fail correctly because of the incorrect arguments passed to `dotnet` commands

2. Change the build definition to actually use .net core SDK 1.0
    After this change the build should finish correctly and unit test results should be published to VSTS

--- a/dotnet-install.ps1
+++ b/dotnet-install.ps1
@@ -1,0 +1,440 @@
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+<#
+.SYNOPSIS
+    Installs dotnet cli
+.DESCRIPTION
+    Installs dotnet cli. If dotnet installation already exists in the given directory
+    it will update it only if the requested version differs from the one already installed.
+.PARAMETER Channel
+    Default: preview
+    Channel is the way of reasoning about stability and quality of dotnet. This parameter takes one of the values:
+    - future - Possibly unstable, frequently changing, may contain new finished and unfinished features
+    - preview - Pre-release stable with known issues and feature gaps
+    - production - Most stable releases
+.PARAMETER Version
+    Default: latest
+    Represents a build version on specific channel. Possible values:
+    - 4-part version in a format A.B.C.D - represents specific version of build
+    - latest - most latest build on specific channel
+    - lkg - last known good version on specific channel
+    Note: LKG work is in progress. Once the work is finished, this will become new default
+.PARAMETER InstallDir
+    Default: %LocalAppData%\Microsoft\dotnet
+    Path to where to install dotnet. Note that binaries will be placed directly in a given directory.
+.PARAMETER Architecture
+    Default: <auto> - this value represents currently running OS architecture
+    Architecture of dotnet binaries to be installed.
+    Possible values are: <auto>, x64 and x86
+.PARAMETER SharedRuntime
+    Default: false
+    Installs just the shared runtime bits, not the entire SDK
+.PARAMETER DebugSymbols
+    If set the installer will include symbols in the installation.
+.PARAMETER DryRun
+    If set it will not perform installation but instead display what command line to use to consistently install
+    currently requested version of dotnet cli. In example if you specify version 'latest' it will display a link
+    with specific version so that this command can be used deterministicly in a build script.
+    It also displays binaries location if you prefer to install or download it yourself.
+.PARAMETER NoPath
+    By default this script will set environment variable PATH for the current process to the binaries folder inside installation folder.
+    If set it will display binaries location but not set any environment variable.
+.PARAMETER Verbose
+    Displays diagnostics information.
+.PARAMETER AzureFeed
+    Default: https://dotnetcli.azureedge.net/dotnet
+    This parameter should not be usually changed by user. It allows to change URL for the Azure feed used by this installer.
+.PARAMETER ProxyAddress
+    If set, the installer will use the proxy when making web requests
+#>
+[cmdletbinding()]
+param(
+   [string]$Channel="rel-1.0.0",
+   [string]$Version="Latest",
+   [string]$InstallDir="<auto>",
+   [string]$Architecture="<auto>",
+   [switch]$SharedRuntime,
+   [switch]$DebugSymbols, # TODO: Switch does not work yet. Symbols zip is not being uploaded yet.
+   [switch]$DryRun,
+   [switch]$NoPath,
+   [string]$AzureFeed="https://dotnetcli.azureedge.net/dotnet",
+   [string]$UncachedFeed="https://dotnetcli.blob.core.windows.net/dotnet",
+   [string]$ProxyAddress
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference="Stop"
+$ProgressPreference="SilentlyContinue"
+
+$BinFolderRelativePath=""
+
+# example path with regex: shared/1.0.0-beta-12345/somepath
+$VersionRegEx="/\d+\.\d+[^/]+/"
+$OverrideNonVersionedFiles=$true
+
+function Say($str) {
+    Write-Host "dotnet-install: $str"
+}
+
+function Say-Verbose($str) {
+    Write-Verbose "dotnet-install: $str"
+}
+
+function Say-Invocation($Invocation) {
+    $command = $Invocation.MyCommand;
+    $args = (($Invocation.BoundParameters.Keys | foreach { "-$_ `"$($Invocation.BoundParameters[$_])`"" }) -join " ")
+    Say-Verbose "$command $args"
+}
+
+function Get-Machine-Architecture() {
+    Say-Invocation $MyInvocation
+
+    # possible values: AMD64, IA64, x86
+    return $ENV:PROCESSOR_ARCHITECTURE
+}
+
+# TODO: Architecture and CLIArchitecture should be unified
+function Get-CLIArchitecture-From-Architecture([string]$Architecture) {
+    Say-Invocation $MyInvocation
+
+    switch ($Architecture.ToLower()) {
+        { $_ -eq "<auto>" } { return Get-CLIArchitecture-From-Architecture $(Get-Machine-Architecture) }
+        { ($_ -eq "amd64") -or ($_ -eq "x64") } { return "x64" }
+        { $_ -eq "x86" } { return "x86" }
+        default { throw "Architecture not supported. If you think this is a bug, please report it at https://github.com/dotnet/cli/issues" }
+    }
+}
+
+function Get-Version-Info-From-Version-Text([string]$VersionText) {
+    Say-Invocation $MyInvocation
+
+    $Data = @($VersionText.Split([char[]]@(), [StringSplitOptions]::RemoveEmptyEntries));
+
+    $VersionInfo = @{}
+    $VersionInfo.CommitHash = $Data[0].Trim()
+    $VersionInfo.Version = $Data[1].Trim()
+    return $VersionInfo
+}
+
+function Load-Assembly([string] $Assembly) {
+    try {
+        Add-Type -Assembly $Assembly | Out-Null
+    }
+    catch {
+        # On Nano Server, Powershell Core Edition is used.  Add-Type is unable to resolve base class assemblies because they are not GAC'd.
+        # Loading the base class assemblies is not unnecessary as the types will automatically get resolved.
+    }
+}
+
+function GetHTTPResponse([Uri] $Uri)
+{
+    $HttpClient = $null
+
+    try {
+        # HttpClient is used vs Invoke-WebRequest in order to support Nano Server which doesn't support the Invoke-WebRequest cmdlet.
+        Load-Assembly -Assembly System.Net.Http
+        if($ProxyAddress){
+            $HttpClientHandler = New-Object System.Net.Http.HttpClientHandler
+            $HttpClientHandler.Proxy =  New-Object System.Net.WebProxy -Property @{Address=$ProxyAddress}
+            $HttpClient = New-Object System.Net.Http.HttpClient -ArgumentList $HttpClientHandler
+        } 
+        else {
+            $HttpClient = New-Object System.Net.Http.HttpClient
+        }
+        # Default timeout for HttpClient is 100s.  For a 50 MB download this assumes 500 KB/s average, any less will time out
+        # 5 minutes allows it to work over much slower connections.
+        $HttpClient.Timeout = New-TimeSpan -Minutes 5 
+        $Response = $HttpClient.GetAsync($Uri).Result
+        if (($Response -eq $null) -or (-not ($Response.IsSuccessStatusCode)))
+        {
+            $ErrorMsg = "Failed to download $Uri."
+            if ($Response -ne $null)
+            {
+                $ErrorMsg += "  $Response"
+            }
+
+            throw $ErrorMsg
+        }
+
+        return $Response
+    }
+    finally {
+        if ($HttpClient -ne $null) {
+            $HttpClient.Dispose()
+        }
+    }
+}
+
+
+function Get-Latest-Version-Info([string]$AzureFeed, [string]$AzureChannel, [string]$CLIArchitecture) {
+    Say-Invocation $MyInvocation
+
+    $VersionFileUrl = $null
+    if ($SharedRuntime) {
+        $VersionFileUrl = "$UncachedFeed/$AzureChannel/dnvm/latest.sharedfx.win.$CLIArchitecture.version"
+    }
+    else {
+        $VersionFileUrl = "$UncachedFeed/Sdk/$AzureChannel/latest.version"
+    }
+    
+    $Response = GetHTTPResponse -Uri $VersionFileUrl
+    $StringContent = $Response.Content.ReadAsStringAsync().Result
+
+    switch ($Response.Content.Headers.ContentType) {
+        { ($_ -eq "application/octet-stream") } { $VersionText = [Text.Encoding]::UTF8.GetString($StringContent) }
+        { ($_ -eq "text/plain") } { $VersionText = $StringContent }
+        default { throw "``$Response.Content.Headers.ContentType`` is an unknown .version file content type." }
+    }
+
+    $VersionInfo = Get-Version-Info-From-Version-Text $VersionText
+
+    return $VersionInfo
+}
+
+# TODO: AzureChannel and Channel should be unified
+function Get-Azure-Channel-From-Channel([string]$Channel) {
+    Say-Invocation $MyInvocation
+
+    # For compatibility with build scripts accept also directly Azure channels names
+    switch ($Channel.ToLower()) {
+        { ($_ -eq "future") -or ($_ -eq "dev") } { return "dev" }
+        #{ $_ -eq "production" } { throw "Production channel does not exist yet" }
+        default { return $_ }
+    }
+}
+
+function Get-Specific-Version-From-Version([string]$AzureFeed, [string]$AzureChannel, [string]$CLIArchitecture, [string]$Version) {
+    Say-Invocation $MyInvocation
+
+    switch ($Version.ToLower()) {
+        { $_ -eq "latest" } {
+            $LatestVersionInfo = Get-Latest-Version-Info -AzureFeed $AzureFeed -AzureChannel $AzureChannel -CLIArchitecture $CLIArchitecture
+            return $LatestVersionInfo.Version
+        }
+        { $_ -eq "lkg" } { throw "``-Version LKG`` not supported yet." }
+        default { return $Version }
+    }
+}
+
+function Get-Download-Links([string]$AzureFeed, [string]$AzureChannel, [string]$SpecificVersion, [string]$CLIArchitecture) {
+    Say-Invocation $MyInvocation
+    
+    $ret = @()
+    
+    if ($SharedRuntime) {
+        $PayloadURL = "$AzureFeed/$AzureChannel/Binaries/$SpecificVersion/dotnet-win-$CLIArchitecture.$SpecificVersion.zip"
+    }
+    else {
+        $PayloadURL = "$AzureFeed/Sdk/$SpecificVersion/dotnet-dev-win-$CLIArchitecture.$SpecificVersion.zip"
+    }
+
+    Say-Verbose "Constructed payload URL: $PayloadURL"
+    $ret += $PayloadURL
+
+    return $ret
+}
+
+function Get-User-Share-Path() {
+    Say-Invocation $MyInvocation
+
+    $InstallRoot = $env:DOTNET_INSTALL_DIR
+    if (!$InstallRoot) {
+        $InstallRoot = "$env:LocalAppData\Microsoft\dotnet"
+    }
+    return $InstallRoot
+}
+
+function Resolve-Installation-Path([string]$InstallDir) {
+    Say-Invocation $MyInvocation
+
+    if ($InstallDir -eq "<auto>") {
+        return Get-User-Share-Path
+    }
+    return $InstallDir
+}
+
+function Get-Version-Info-From-Version-File([string]$InstallRoot, [string]$RelativePathToVersionFile) {
+    Say-Invocation $MyInvocation
+
+    $VersionFile = Join-Path -Path $InstallRoot -ChildPath $RelativePathToVersionFile
+    Say-Verbose "Local version file: $VersionFile"
+    
+    if (Test-Path $VersionFile) {
+        $VersionText = cat $VersionFile
+        Say-Verbose "Local version file text: $VersionText"
+        return Get-Version-Info-From-Version-Text $VersionText
+    }
+
+    Say-Verbose "Local version file not found."
+
+    return $null
+}
+
+function Is-Dotnet-Package-Installed([string]$InstallRoot, [string]$RelativePathToPackage, [string]$SpecificVersion) {
+    Say-Invocation $MyInvocation
+    
+    $DotnetPackagePath = Join-Path -Path $InstallRoot -ChildPath $RelativePathToPackage | Join-Path -ChildPath $SpecificVersion
+    Say-Verbose "Is-Dotnet-Package-Installed: Path to a package: $DotnetPackagePath"
+    return Test-Path $DotnetPackagePath -PathType Container
+}
+
+function Get-Absolute-Path([string]$RelativeOrAbsolutePath) {
+    # Too much spam
+    # Say-Invocation $MyInvocation
+
+    return $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($RelativeOrAbsolutePath)
+}
+
+function Get-Path-Prefix-With-Version($path) {
+    $match = [regex]::match($path, $VersionRegEx)
+    if ($match.Success) {
+        return $entry.FullName.Substring(0, $match.Index + $match.Length)
+    }
+    
+    return $null
+}
+
+function Get-List-Of-Directories-And-Versions-To-Unpack-From-Dotnet-Package([System.IO.Compression.ZipArchive]$Zip, [string]$OutPath) {
+    Say-Invocation $MyInvocation
+    
+    $ret = @()
+    foreach ($entry in $Zip.Entries) {
+        $dir = Get-Path-Prefix-With-Version $entry.FullName
+        if ($dir -ne $null) {
+            $path = Get-Absolute-Path $(Join-Path -Path $OutPath -ChildPath $dir)
+            if (-Not (Test-Path $path -PathType Container)) {
+                $ret += $dir
+            }
+        }
+    }
+    
+    $ret = $ret | Sort-Object | Get-Unique
+    
+    $values = ($ret | foreach { "$_" }) -join ";"
+    Say-Verbose "Directories to unpack: $values"
+    
+    return $ret
+}
+
+# Example zip content and extraction algorithm:
+# Rule: files if extracted are always being extracted to the same relative path locally
+# .\
+#       a.exe   # file does not exist locally, extract
+#       b.dll   # file exists locally, override only if $OverrideFiles set
+#       aaa\    # same rules as for files
+#           ...
+#       abc\1.0.0\  # directory contains version and exists locally
+#           ...     # do not extract content under versioned part
+#       abc\asd\    # same rules as for files
+#            ...
+#       def\ghi\1.0.1\  # directory contains version and does not exist locally
+#           ...         # extract content
+function Extract-Dotnet-Package([string]$ZipPath, [string]$OutPath) {
+    Say-Invocation $MyInvocation
+
+    Load-Assembly -Assembly System.IO.Compression.FileSystem
+    Set-Variable -Name Zip
+    try {
+        $Zip = [System.IO.Compression.ZipFile]::OpenRead($ZipPath)
+        
+        $DirectoriesToUnpack = Get-List-Of-Directories-And-Versions-To-Unpack-From-Dotnet-Package -Zip $Zip -OutPath $OutPath
+        
+        foreach ($entry in $Zip.Entries) {
+            $PathWithVersion = Get-Path-Prefix-With-Version $entry.FullName
+            if (($PathWithVersion -eq $null) -Or ($DirectoriesToUnpack -contains $PathWithVersion)) {
+                $DestinationPath = Get-Absolute-Path $(Join-Path -Path $OutPath -ChildPath $entry.FullName)
+                $DestinationDir = Split-Path -Parent $DestinationPath
+                $OverrideFiles=$OverrideNonVersionedFiles -Or (-Not (Test-Path $DestinationPath))
+                if ((-Not $DestinationPath.EndsWith("\")) -And $OverrideFiles) {
+                    New-Item -ItemType Directory -Force -Path $DestinationDir | Out-Null
+                    [System.IO.Compression.ZipFileExtensions]::ExtractToFile($entry, $DestinationPath, $OverrideNonVersionedFiles)
+                }
+            }
+        }
+    }
+    finally {
+        if ($Zip -ne $null) {
+            $Zip.Dispose()
+        }
+    }
+}
+
+function DownloadFile([Uri]$Uri, [string]$OutPath) {
+    $Stream = $null
+
+    try {
+        $Response = GetHTTPResponse -Uri $Uri
+        $Stream = $Response.Content.ReadAsStreamAsync().Result
+        $File = [System.IO.File]::Create($OutPath)
+        $Stream.CopyTo($File)
+        $File.Close()
+    }
+    finally {
+        if ($Stream -ne $null) {
+            $Stream.Dispose()
+        }
+    }
+}
+
+function Prepend-Sdk-InstallRoot-To-Path([string]$InstallRoot, [string]$BinFolderRelativePath) {
+    $BinPath = Get-Absolute-Path $(Join-Path -Path $InstallRoot -ChildPath $BinFolderRelativePath)
+    if (-Not $NoPath) {
+        Say "Adding to current process PATH: `"$BinPath`". Note: This change will not be visible if PowerShell was run as a child process."
+        $env:path = "$BinPath;" + $env:path
+    }
+    else {
+        Say "Binaries of dotnet can be found in $BinPath"
+    }
+}
+
+$AzureChannel = Get-Azure-Channel-From-Channel -Channel $Channel
+$CLIArchitecture = Get-CLIArchitecture-From-Architecture $Architecture
+$SpecificVersion = Get-Specific-Version-From-Version -AzureFeed $AzureFeed -AzureChannel $AzureChannel -CLIArchitecture $CLIArchitecture -Version $Version
+$DownloadLinks = Get-Download-Links -AzureFeed $AzureFeed -AzureChannel $AzureChannel -SpecificVersion $SpecificVersion -CLIArchitecture $CLIArchitecture
+
+if ($DryRun) {
+    Say "Payload URLs:"
+    foreach ($DownloadLink in $DownloadLinks) {
+        Say "- $DownloadLink"
+    }
+    Say "Repeatable invocation: .\$($MyInvocation.MyCommand) -Version $SpecificVersion -Channel $Channel -Architecture $CLIArchitecture -InstallDir $InstallDir"
+    exit 0
+}
+
+$InstallRoot = Resolve-Installation-Path $InstallDir
+Say-Verbose "InstallRoot: $InstallRoot"
+
+$IsSdkInstalled = Is-Dotnet-Package-Installed -InstallRoot $InstallRoot -RelativePathToPackage "sdk" -SpecificVersion $SpecificVersion
+Say-Verbose ".NET SDK installed? $IsSdkInstalled"
+if ($IsSdkInstalled) {
+    Say ".NET SDK version $SpecificVersion is already installed."
+    Prepend-Sdk-InstallRoot-To-Path -InstallRoot $InstallRoot -BinFolderRelativePath $BinFolderRelativePath
+    exit 0
+}
+
+New-Item -ItemType Directory -Force -Path $InstallRoot | Out-Null
+
+$free = Get-CimInstance -Class win32_logicaldisk | where Deviceid -eq "$((Get-Item $InstallRoot).PSDrive.Name):"
+if ($free.Freespace / 1MB -le 250 ) {
+    Say "there is not enough disk space on drive c:"
+    exit 0
+}
+
+foreach ($DownloadLink in $DownloadLinks) {
+    $ZipPath = [System.IO.Path]::GetTempFileName()
+    Say "Downloading $DownloadLink"
+    DownloadFile -Uri $DownloadLink -OutPath $ZipPath
+
+    Say "Extracting zip from $DownloadLink"
+    Extract-Dotnet-Package -ZipPath $ZipPath -OutPath $InstallRoot
+
+    Remove-Item $ZipPath
+}
+
+Prepend-Sdk-InstallRoot-To-Path -InstallRoot $InstallRoot -BinFolderRelativePath $BinFolderRelativePath
+
+Say "Installation finished"
+exit 0

--- a/dotnet-install.sh
+++ b/dotnet-install.sh
@@ -1,0 +1,688 @@
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+# Note: This script should be compatible with the dash shell used in Ubuntu. So avoid bashisms! See https://wiki.ubuntu.com/DashAsBinSh for more info
+
+# Stop script on NZEC
+set -e
+# Stop script if unbound variable found (use ${var:-} if intentional)
+set -u
+# By default cmd1 | cmd2 returns exit code of cmd2 regardless of cmd1 success
+# This is causing it to fail
+set -o pipefail
+
+# Use in the the functions: eval $invocation
+invocation='say_verbose "Calling: ${yellow:-}${FUNCNAME[0]} ${green:-}$*${normal:-}"'
+
+# standard output may be used as a return value in the functions
+# we need a way to write text on the screen in the functions so that
+# it won't interfere with the return value.
+# Exposing stream 3 as a pipe to standard output of the script itself
+exec 3>&1
+
+# Setup some colors to use. These need to work in fairly limited shells, like the Ubuntu Docker container where there are only 8 colors.
+# See if stdout is a terminal
+if [ -t 1 ]; then
+    # see if it supports colors
+    ncolors=$(tput colors)
+    if [ -n "$ncolors" ] && [ $ncolors -ge 8 ]; then
+        bold="$(tput bold       || echo)"
+        normal="$(tput sgr0     || echo)"
+        black="$(tput setaf 0   || echo)"
+        red="$(tput setaf 1     || echo)"
+        green="$(tput setaf 2   || echo)"
+        yellow="$(tput setaf 3  || echo)"
+        blue="$(tput setaf 4    || echo)"
+        magenta="$(tput setaf 5 || echo)"
+        cyan="$(tput setaf 6    || echo)"
+        white="$(tput setaf 7   || echo)"
+    fi
+fi
+
+say_err() {
+    printf "%b\n" "${red:-}dotnet_install: Error: $1${normal:-}" >&2
+}
+
+say() {
+    # using stream 3 (defined in the beginning) to not interfere with stdout of functions
+    # which may be used as return value
+    printf "%b\n" "${cyan:-}dotnet-install:${normal:-} $1" >&3
+}
+
+say_verbose() {
+    if [ "$verbose" = true ]; then
+        say "$1"
+    fi
+}
+
+get_current_os_name() {
+    eval $invocation
+
+    local uname=$(uname)
+    if [ "$uname" = "Darwin" ]; then
+        echo "osx"
+        return 0
+    else
+        if [ -e /etc/os-release ]; then
+            . /etc/os-release
+
+            case "$ID.$VERSION_ID" in
+                "centos.7")
+                    echo "centos"
+                    return 0
+                    ;;
+                "debian.8")
+                    echo "debian"
+                    return 0
+                    ;;
+                "fedora.23")
+                    echo "fedora.23"
+                    return 0
+                    ;;
+                "fedora.24")
+                    echo "fedora.24"
+                    return 0
+                    ;;
+                "opensuse.13.2")
+                    echo "opensuse.13.2"
+                    return 0
+                    ;;
+                "opensuse.42.1")
+                    echo "opensuse.42.1"
+                    return 0
+                    ;;
+                "rhel.7.0" | "rhel.7.1" | "rhel.7.2")
+                    echo "rhel"
+                    return 0
+                    ;;
+                "ubuntu.14.04")
+                    echo "ubuntu"
+                    return 0
+                    ;;
+                "ubuntu.16.04")
+                    echo "ubuntu.16.04"
+                    return 0
+                    ;;
+                "ubuntu.16.10")
+                    echo "ubuntu.16.10"
+                    return 0
+                    ;;
+                "alpine.3.4.3")
+                    echo "alpine"
+                    return 0
+                    ;;
+            esac
+        fi
+    fi
+    
+    say_err "OS name could not be detected: $ID.$VERSION_ID"
+    return 1
+}
+
+machine_has() {
+    eval $invocation
+    
+    which "$1" > /dev/null 2>&1
+    return $?
+}
+
+check_min_reqs() {
+    if ! machine_has "curl"; then
+        say_err "curl is required to download dotnet. Install curl to proceed."
+        return 1
+    fi
+    
+    return 0
+}
+
+check_pre_reqs() {
+    eval $invocation
+    
+    local failing=false;
+
+    if [ "${DOTNET_INSTALL_SKIP_PREREQS:-}" = "1" ]; then
+        return 0
+    fi
+
+    if [ "$(uname)" = "Linux" ]; then
+        if ! [ -x "$(command -v ldconfig)" ]; then
+            echo "ldconfig is not in PATH, trying /sbin/ldconfig."
+            LDCONFIG_COMMAND="/sbin/ldconfig"
+        else
+            LDCONFIG_COMMAND="ldconfig"
+        fi
+
+        [ -z "$($LDCONFIG_COMMAND -p | grep libunwind)" ] && say_err "Unable to locate libunwind. Install libunwind to continue" && failing=true
+        [ -z "$($LDCONFIG_COMMAND -p | grep libssl)" ] && say_err "Unable to locate libssl. Install libssl to continue" && failing=true
+        [ -z "$($LDCONFIG_COMMAND -p | grep libcurl)" ] && say_err "Unable to locate libcurl. Install libcurl to continue" && failing=true
+        [ -z "$($LDCONFIG_COMMAND -p | grep libicu)" ] && say_err "Unable to locate libicu. Install libicu to continue" && failing=true
+    fi
+
+    if [ "$failing" = true ]; then
+       return 1
+    fi
+    
+    return 0
+}
+
+# args:
+# input - $1
+to_lowercase() {
+    #eval $invocation
+    
+    echo "$1" | tr '[:upper:]' '[:lower:]'
+    return 0
+}
+
+# args:
+# input - $1
+remove_trailing_slash() {
+    #eval $invocation
+    
+    local input=${1:-}
+    echo "${input%/}"
+    return 0
+}
+
+# args:
+# input - $1
+remove_beginning_slash() {
+    #eval $invocation
+    
+    local input=${1:-}
+    echo "${input#/}"
+    return 0
+}
+
+# args:
+# root_path - $1
+# child_path - $2 - this parameter can be empty
+combine_paths() {
+    eval $invocation
+    
+    # TODO: Consider making it work with any number of paths. For now:
+    if [ ! -z "${3:-}" ]; then
+        say_err "combine_paths: Function takes two parameters."
+        return 1
+    fi
+    
+    local root_path=$(remove_trailing_slash $1)
+    local child_path=$(remove_beginning_slash ${2:-})
+    say_verbose "combine_paths: root_path=$root_path"
+    say_verbose "combine_paths: child_path=$child_path"
+    echo "$root_path/$child_path"
+    return 0
+}
+
+get_machine_architecture() {
+    eval $invocation
+    
+    # Currently the only one supported
+    echo "x64"
+    return 0
+}
+
+# args:
+# architecture - $1
+get_normalized_architecture_from_architecture() {
+    eval $invocation
+    
+    local architecture=$(to_lowercase $1)
+    case $architecture in
+        \<auto\>)
+            echo "$(get_normalized_architecture_from_architecture $(get_machine_architecture))"
+            return 0
+            ;;
+        amd64|x64)
+            echo "x64"
+            return 0
+            ;;
+        x86)
+            say_err "Architecture ``x86`` currently not supported"
+            return 1
+            ;;
+    esac
+   
+    say_err "Architecture ``$architecture`` not supported. If you think this is a bug, please report it at https://github.com/dotnet/cli/issues"
+    return 1
+}
+
+# version_info is a conceptual two line string representing commit hash and 4-part version
+# format:
+# Line 1: # commit_hash
+# Line 2: # 4-part version
+
+# args:
+# version_text - stdin
+get_version_from_version_info() {
+    eval $invocation
+    
+    cat | tail -n 1
+    return 0
+}
+
+# args:
+# version_text - stdin
+get_commit_hash_from_version_info() {
+    eval $invocation
+    
+    cat | head -n 1
+    return 0
+}
+
+# args:
+# install_root - $1
+# relative_path_to_package - $2
+# specific_version - $3
+is_dotnet_package_installed() {
+    eval $invocation
+    
+    local install_root=$1
+    local relative_path_to_package=$2
+    local specific_version=${3//[$'\t\r\n']}
+    
+    local dotnet_package_path=$(combine_paths $(combine_paths $install_root $relative_path_to_package) $specific_version)
+    say_verbose "is_dotnet_package_installed: dotnet_package_path=$dotnet_package_path"
+    
+    if [ -d "$dotnet_package_path" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# args:
+# azure_feed - $1
+# azure_channel - $2
+# normalized_architecture - $3
+get_latest_version_info() {
+    eval $invocation
+    
+    local azure_feed=$1
+    local azure_channel=$2
+    local normalized_architecture=$3
+    
+    local osname
+    osname=$(get_current_os_name) || return 1
+
+    local version_file_url=null
+    if [ "$shared_runtime" = true ]; then
+        version_file_url="$uncached_feed/$azure_channel/dnvm/latest.sharedfx.$osname.$normalized_architecture.version"
+    else
+        version_file_url="$uncached_feed/Sdk/$azure_channel/latest.version"
+    fi
+    say_verbose "get_latest_version_info: latest url: $version_file_url"
+    
+    download $version_file_url
+    return $?
+}
+
+# args:
+# channel - $1
+get_azure_channel_from_channel() {
+    eval $invocation
+    
+    local channel=$(to_lowercase $1)
+    case $channel in
+        future|dev)
+            echo "dev"
+            return 0
+            ;;
+#        production)
+#            say_err "Production channel does not exist yet"
+#            return 1
+    esac
+    
+    echo $channel
+    return 0
+}
+
+# args:
+# azure_feed - $1
+# azure_channel - $2
+# normalized_architecture - $3
+# version - $4
+get_specific_version_from_version() {
+    eval $invocation
+    
+    local azure_feed=$1
+    local azure_channel=$2
+    local normalized_architecture=$3
+    local version=$(to_lowercase $4)
+
+    case $version in
+        latest)
+            local version_info
+        version_info="$(get_latest_version_info $azure_feed $azure_channel $normalized_architecture)" || return 1
+            say_verbose "get_specific_version_from_version: version_info=$version_info"
+            echo "$version_info" | get_version_from_version_info
+            return 0
+            ;;
+        lkg)
+            say_err "``--version LKG`` not supported yet."
+            return 1
+            ;;
+        *)
+            echo $version
+            return 0
+            ;;
+    esac
+}
+
+# args:
+# azure_feed - $1
+# azure_channel - $2
+# normalized_architecture - $3
+# specific_version - $4
+construct_download_link() {
+    eval $invocation
+    
+    local azure_feed=$1
+    local azure_channel=$2
+    local normalized_architecture=$3
+    local specific_version=${4//[$'\t\r\n']}
+    
+    local osname
+    osname=$(get_current_os_name) || return 1
+    
+    local download_link=null
+    if [ "$shared_runtime" = true ]; then
+        download_link="$azure_feed/$azure_channel/Binaries/$specific_version/dotnet-$osname-$normalized_architecture.$specific_version.tar.gz"
+    else
+        download_link="$azure_feed/Sdk/$specific_version/dotnet-dev-$osname-$normalized_architecture.$specific_version.tar.gz"
+    fi
+    
+    echo "$download_link"
+    return 0
+}
+
+get_user_share_path() {
+    eval $invocation
+    
+    if [ ! -z "${DOTNET_INSTALL_DIR:-}" ]; then
+        echo $DOTNET_INSTALL_DIR
+    else
+        echo "$HOME/.dotnet"
+    fi
+    return 0
+}
+
+# args:
+# install_dir - $1
+resolve_installation_path() {
+    eval $invocation
+    
+    local install_dir=$1
+    if [ "$install_dir" = "<auto>" ]; then
+        local user_share_path=$(get_user_share_path)
+        say_verbose "resolve_installation_path: share_path=$user_share_path"
+        echo "$user_share_path"
+        return 0
+    fi
+    
+    echo "$install_dir"
+    return 0
+}
+
+# args:
+# install_root - $1
+get_installed_version_info() {
+    eval $invocation
+    
+    local install_root=$1
+    local version_file=$(combine_paths "$install_root" "$local_version_file_relative_path")
+    say_verbose "Local version file: $version_file"
+    if [ ! -z "$version_file" ] | [ -r "$version_file" ]; then
+        local version_info="$(cat $version_file)"
+        echo "$version_info"
+        return 0
+    fi
+    
+    say_verbose "Local version file not found."
+    return 0
+}
+
+# args:
+# relative_or_absolute_path - $1
+get_absolute_path() {
+    eval $invocation
+    
+    local relative_or_absolute_path=$1
+    echo $(cd $(dirname "$1") && pwd -P)/$(basename "$1")
+    return 0
+}
+
+# args:
+# input_files - stdin
+# root_path - $1
+# out_path - $2
+# override - $3
+copy_files_or_dirs_from_list() {
+    eval $invocation
+
+    local root_path=$(remove_trailing_slash $1)
+    local out_path=$(remove_trailing_slash $2)
+    local override=$3
+    local override_switch=$(if [ "$override" = false ]; then printf -- "-n"; fi)
+    
+    cat | uniq | while read -r file_path; do
+        local path=$(remove_beginning_slash ${file_path#$root_path})
+        local target=$out_path/$path
+        if [ "$override" = true ] || (! ([ -d "$target" ] || [ -e "$target" ])); then
+            mkdir -p $out_path/$(dirname $path)
+            cp -R $override_switch $root_path/$path $target
+        fi
+    done
+}
+
+# args:
+# zip_path - $1
+# out_path - $2
+extract_dotnet_package() {
+    eval $invocation
+    
+    local zip_path=$1
+    local out_path=$2
+    
+    local temp_out_path=$(mktemp -d $temporary_file_template)
+    
+    local failed=false
+    tar -xzf "$zip_path" -C "$temp_out_path" > /dev/null || failed=true
+    
+    local folders_with_version_regex='^.*/[0-9]+\.[0-9]+[^/]+/'
+    find $temp_out_path -type f | grep -Eo $folders_with_version_regex | copy_files_or_dirs_from_list $temp_out_path $out_path false
+    find $temp_out_path -type f | grep -Ev $folders_with_version_regex | copy_files_or_dirs_from_list $temp_out_path $out_path true
+    
+    rm -rf $temp_out_path
+    
+    if [ "$failed" = true ]; then
+        say_err "Extraction failed"
+        return 1
+    fi
+}
+
+# args:
+# remote_path - $1
+# [out_path] - $2 - stdout if not provided
+download() {
+    eval $invocation
+    
+    local remote_path=$1
+    local out_path=${2:-}
+
+    local failed=false
+    if [ -z "$out_path" ]; then
+        curl --fail -s $remote_path || failed=true
+    else
+        curl --fail -s -o $out_path $remote_path || failed=true
+    fi
+    
+    if [ "$failed" = true ]; then
+        say_err "Download failed"
+        return 1
+    fi
+}
+
+calculate_vars() {
+    eval $invocation
+    
+    azure_channel=$(get_azure_channel_from_channel "$channel")
+    say_verbose "azure_channel=$azure_channel"
+    
+    normalized_architecture=$(get_normalized_architecture_from_architecture "$architecture")
+    say_verbose "normalized_architecture=$normalized_architecture"
+    
+    specific_version=$(get_specific_version_from_version $azure_feed $azure_channel $normalized_architecture $version)
+    say_verbose "specific_version=$specific_version"
+    if [ -z "$specific_version" ]; then
+        say_err "Could not get version information."
+        return 1
+    fi
+    
+    download_link=$(construct_download_link $azure_feed $azure_channel $normalized_architecture $specific_version)
+    say_verbose "download_link=$download_link"
+    
+    install_root=$(resolve_installation_path $install_dir)
+    say_verbose "install_root=$install_root"
+}
+
+install_dotnet() {
+    eval $invocation
+    
+    if is_dotnet_package_installed $install_root "sdk" $specific_version; then
+        say ".NET SDK version $specific_version is already installed."
+        return 0
+    fi
+    
+    mkdir -p $install_root
+    zip_path=$(mktemp $temporary_file_template)
+    say_verbose "Zip path: $zip_path"
+    
+    say "Downloading $download_link"
+    download "$download_link" $zip_path
+    say_verbose "Downloaded file exists and readable? $(if [ -r $zip_path ]; then echo "yes"; else echo "no"; fi)"
+    
+    say "Extracting zip"
+    extract_dotnet_package $zip_path $install_root
+    
+    return 0
+}
+
+local_version_file_relative_path="/.version"
+bin_folder_relative_path=""
+temporary_file_template="${TMPDIR:-/tmp}/dotnet.XXXXXXXXX"
+
+channel="rel-1.0.0"
+version="Latest"
+install_dir="<auto>"
+architecture="<auto>"
+debug_symbols=false
+dry_run=false
+no_path=false
+azure_feed="https://dotnetcli.azureedge.net/dotnet"
+uncached_feed="https://dotnetcli.blob.core.windows.net/dotnet"
+verbose=false
+shared_runtime=false
+
+while [ $# -ne 0 ]
+do
+    name=$1
+    case $name in
+        -c|--channel|-[Cc]hannel)
+            shift
+            channel=$1
+            ;;
+        -v|--version|-[Vv]ersion)
+            shift
+            version="$1"
+            ;;
+        -i|--install-dir|-[Ii]nstall[Dd]ir)
+            shift
+            install_dir="$1"
+            ;;
+        --arch|--architecture|-[Aa]rch|-[Aa]rchitecture)
+            shift
+            architecture="$1"
+            ;;
+        --shared-runtime|-[Ss]hared[Rr]untime)
+            shared_runtime=true
+            ;;
+        --debug-symbols|-[Dd]ebug[Ss]ymbols)
+            debug_symbols=true
+            ;;
+        --dry-run|-[Dd]ry[Rr]un)
+            dry_run=true
+            ;;
+        --no-path|-[Nn]o[Pp]ath)
+            no_path=true
+            ;;
+        --verbose|-[Vv]erbose)
+            verbose=true
+            ;;
+        --azure-feed|-[Aa]zure[Ff]eed)
+            shift
+            azure_feed="$1"
+            ;;
+        -?|--?|-h|--help|-[Hh]elp)
+            script_name="$(basename $0)"
+            echo ".NET Tools Installer"
+            echo "Usage: $script_name [-c|--channel <CHANNEL>] [-v|--version <VERSION>] [-p|--prefix <DESTINATION>]"
+            echo "       $script_name -h|-?|--help"
+            echo ""
+            echo "$script_name is a simple command line interface for obtaining dotnet cli."
+            echo ""
+            echo "Options:"
+            echo "  -c,--channel <CHANNEL>         Download from the CHANNEL specified (default: $channel)."
+            echo "      -Channel"
+            echo "  -v,--version <VERSION>         Use specific version, ``latest`` or ``lkg``. Defaults to ``latest``."
+            echo "      -Version"
+            echo "  -i,--install-dir <DIR>         Install under specified location (see Install Location below)"
+            echo "      -InstallDir"
+            echo "  --architecture <ARCHITECTURE>  Architecture of .NET Tools. Currently only x64 is supported."
+            echo "      --arch,-Architecture,-Arch"
+            echo "  --shared-runtime               Installs just the shared runtime bits, not the entire SDK."
+            echo "      -SharedRuntime"
+            echo "  --debug-symbols,-DebugSymbols  Specifies if symbols should be included in the installation."
+            echo "  --dry-run,-DryRun              Do not perform installation. Display download link."
+            echo "  --no-path, -NoPath             Do not set PATH for the current process."
+            echo "  --verbose,-Verbose             Display diagnostics information."
+            echo "  --azure-feed,-AzureFeed        Azure feed location. Defaults to $azure_feed"
+            echo "  -?,--?,-h,--help,-Help         Shows this help message"
+            echo ""
+            echo "Install Location:"
+            echo "  Location is chosen in following order:"
+            echo "    - --install-dir option"
+            echo "    - Environmental variable DOTNET_INSTALL_DIR"
+            echo "    - /usr/local/share/dotnet"
+            exit 0
+            ;;
+        *)
+            say_err "Unknown argument \`$name\`"
+            exit 1
+            ;;
+    esac
+
+    shift
+done
+
+check_min_reqs
+calculate_vars
+if [ "$dry_run" = true ]; then
+    say "Payload URL: $download_link"
+    say "Repeatable invocation: ./$(basename $0) --version $specific_version --channel $channel --install-dir $install_dir"
+    exit 0
+fi
+
+check_pre_reqs
+install_dotnet
+
+bin_path=$(get_absolute_path $(combine_paths $install_root $bin_folder_relative_path))
+if [ "$no_path" = false ]; then
+    say "Adding to current process PATH: ``$bin_path``. Note: This change will be visible only when sourcing script."
+    export PATH=$bin_path:$PATH
+else
+    say "Binaries of dotnet can be found in $bin_path"
+fi
+
+say "Installation finished successfully."

